src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -710,14 +710,20 @@ theorem or_comm (x y : BitVec w) :
   simp [Bool.or_comm]
 instance : Std.Commutative (fun (x y : BitVec w) => x ||| y) := ⟨BitVec.or_comm⟩
 
-@[simp] theorem or_self {x : BitVec w} : x ||| x = x := by
+@[simp] theorem or_self (x : BitVec w) : x ||| x = x := by
   ext i
   simp
 
-@[simp] theorem or_zero {x : BitVec w} : x ||| 0#w = x := by
+instance : Std.IdempotentOp (α := BitVec n) (· ||| · ) where
+  idempotent := BitVec.or_self
```

Comment:
Setting up Lawful classes is not sufficient reason to change the implicitness. You can use, e.g.
```
instance : Std.IdempotentOp (α := BitVec n) (· ||| · ) where
  idempotent _ := BitVec.or_self
```
here with the original implicitness.

Reply:
OK, that makes sense.